### ISSUE TITLE
Implement Age 5 cards for Artifacts Expansion

### ISSUE THREAD

**Issue by micahstairs** [micahstairs/bga-innovation#128]
This issue will track the implementation of all Age 5 cards for the Artifacts expansion. Please update this comment before working on any of these cards (if the work is unblocked) so that we don't have duplicate work being done. We no longer need to file a separate issue for each card, since a consolidated list of the cards will keep us more organized.

  - [x] **155 - Boerhavve Silver Microscope**
  - [x] **156 - Principia**
  - [x] **157 - Bill of Rights**
  - [x] **158 - Ship of the Line Sussex**
  - [x] **159 - Barque-Longue La Belle**
  - [x] **160 - Hudson's Bay Company Archives**
  - [x] **161 - Gujin Tushu Jinsheng**
  - [x] **162 - The Daily Courant**
  - [x] **163 - Sandham Room Cricket Bat**
  - [x] **164 - Almira, Queen of the Castle**

**Comment by ultimatefiend** [comment]
this is what I had to this point:

       // id 158, Artifacts age 5: Ship of the Line Sussex
        case "158N1A":
            //If you have no cards in your score pile, choose a color 
            $number_of_cards_in_score_pile = self::countCardsInLocation($player_id, 'score');
            // and score all cards of that color from your board. 
            if ($number_of_cards_in_score_pile == 0)
            {
                $options = array(
                    'player_id' => $player_id,
                    'can_pass' => false,
                    'auxiliary_value' => 1,
                    
                    'choose_color' => true
                );
            }
            else {
                // Otherwise, return all cards from your score pile.
                $options = array(
                    'player_id' => $player_id,
                    'can_pass' => false,
                    'auxiliary_value' => 0,
                    
                    'owner_from' => $player_id,
                    'location_from' => 'score',
                    'owner_to' => 0,
                    'location_to' => 'deck'
                );
            }
                
            break;
            
        case "158N1B":
            // and score all cards of that color from your board. 
            $options = array(
                'player_id' => $player_id,
                'can_pass' => false,
                
                'color' => array(self::getGameStateValue('auxiliary_value')), // choice is a color
                
                'owner_from' => $player_id,
                'location_from' => 'board',
                'owner_to' => $player_id,
                'location_to' => 'score',
                'score_keyword' => true
            );            
                
            break;
...            
                        // id 158, Artifacts age 5: Ship of the Line Sussex
            case "158N1A":
                $message_for_player = clienttranslate('${You} must choose a color');
                $message_for_others = clienttranslate('${player_name} must choose a color');
                break;
...

            // id 158, Artifacts age 5: Ship of the Line Sussex
            case "158N1":
                $number_of_cards_in_score_pile = self::countCardsInLocation($player_id, 'score');
                // and score all cards of that color from your board. 
                if ($number_of_cards_in_score_pile == 0)
                {
                    $step_max = 2;
                }
                else {
                    $step_max = 1;
                }
                break;
...

            // id 158, Artifacts age 5: Ship of the Line Sussex
            case "158N1A":
                if (self::getGameStateValue('auxiliary_value') == 1) {
                    self::notifyPlayer($player_id, 'log', clienttranslate('${You} choose ${color}.'), array('i18n' => array('color'), 'You' => 'You', 'color' => self::getColorInClear($choice)));
                    self::notifyAllPlayersBut($player_id, 'log', clienttranslate('${player_name} chooses ${color}.'), array('i18n' => array('color'), 'player_name' => self::getColoredText(self::getPlayerNameFromId($player_id), $player_id), 'color' => self::getColorInClear($choice)));
                    self::setGameStateValue('auxiliary_value', $choice);
                }
                else {
                    
                }
                break;

**Comment by ultimatefiend** [comment]
and this in dbmodel.sql:

(158, 1, 5, 1, 5, 0, 5, 5, 5, FALSE, 0, 'deck', 0, NULL, FALSE),

**Comment by micahstairs** [comment]
Thanks @ultimatefiend! I can take a stab at this with what you have so far.

**Comment by micahstairs** [comment]
So for Ship of the Line Sussex, we can avoid the auxiliary_value problem by using a different range of values to represent the choice type which doesn't overlap with the colors. For example, we can use 5 to represent that cards were returned instead of a color being chosen. However, there's another problem in `stInterSelectionMove` where we want the default case to be executed if cards were returned, and if we add a case for "158N1A" to handle when a color is chosen then the default case won't be executed. To get around this problem, I'm going to try splitting the different types of interactions into different steps.
